In [1]:
import pandas as pd
import scipy
from scipy import stats
import numpy as np

Hypothesis Testing Exercise, A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units, at 5% significance level.

Its a 2 sample 2 t-test, as we are comparing 2 units mean(diameter mean of cutlets from 2 different units) Defining hypothesis test: Ho, mu(unit A) = mu(unit B) Ha, mu(unit A) != mu(unit B)

In [3]:
df_cutlets= pd.read_csv("D:\Ankita\Assignments\\Cutlets.csv")
df_cutlets.info()
df_cutlets[df_cutlets.duplicated()].shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


(0, 2)

In [4]:
arr_unit1=df_cutlets['Unit A'].values
arr_unit2=df_cutlets['Unit B'].values

stats.ttest_ind(arr_unit1,arr_unit2)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

significance level mentioned 0.05, translates to 95% of confidence level.

result:
(p-value)0.472 > 0.05(alfa)

its a Null hypothesis, Ho: mu(unit A) = mu(unit B), statistically no significant difference in the diameter of cutlets from Unit A and Unit B.